src/test/scala/com/mongodb/spark/rdd/partitioner/MongoSplitVectorPartitionerSpec.scala

```diff
@@ -46,6 +48,34 @@ class MongoSplitVectorPartitionerSpec extends RequiresMongoDB {
       readConfig.copy(partitionerOptions = Map("partitionSizeMB" -> "10")), pipeline
     ).length shouldBe 1
   }
+
+  it should "use the provided pipeline for min and max keys" in {
+    loadSampleData(10)
+
+    val readConf = readConfig.copy(partitionerOptions = Map("partitionSizeMB" -> "1"))
+    val rangePipeline = Array(BsonDocument.parse(s"""{$$match: { $partitionKey: {$$gte: "00001", $$lt: "00022"}}}"""))
+
+    val rangePartitions = MongoSplitVectorPartitioner.partitions(mongoConnector, readConf, rangePipeline)
+    getQueryBoundForKey(rangePartitions.head, "$gte") should equal("00001")
+    getQueryBoundForKey(rangePartitions.reverse.head, "$lt") should equal("00022")
+    rangePartitions.length should (be >= 4 and be <= 5)
+  }
+
+  it should "use the users pipeline when set in a rdd / dataframe" in {
+    loadSampleData(10)
+
+    val readConf = readConfig.copy(partitioner = MongoSplitVectorPartitioner, partitionerOptions = Map("partitionSizeMB" -> "1"))
+    val rangePipeline = BsonDocument.parse(s"""{$$match: { $partitionKey: {$$gte: "00001", $$lt: "00031"}}}""")
+
+    val sparkSession = SparkSession.builder().getOrCreate()
+    val rdd = MongoSpark.load(sparkSession.sparkContext, readConf).withPipeline(Seq(rangePipeline))
+    rdd.count() should equal(30)
+    rdd.partitions.length should (be >= 6 and be <= 7)
```

Comment:
Thats what it does test as length is an int.  I think it was just ease of use with the scalatest spec syntax.